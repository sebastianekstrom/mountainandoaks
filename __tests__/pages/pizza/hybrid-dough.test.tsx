import React from "react";
import { render, screen } from "@testing-library/react";
import HybridDough from "pages/pizza/hybrid-dough";

describe("HybridDough page", () => {
  it("renders title correctly", () => {
    render(<HybridDough />);
    const title = screen.queryAllByText(/The hybrid pizza dough/i);
    expect(title).toBeDefined();
  });

  it("renders the ingress text", () => {
    render(<HybridDough />);
    const ingress = screen.queryAllByText(
      /This dough is the result of me experimenting/i
    );
    expect(ingress).toBeDefined();
  });

  it("renders the correct ingredients", () => {
    render(<HybridDough />);
    const ingredientList = screen.getByTestId("INGREDIENT_LIST");
    expect(ingredientList).toMatchSnapshot();
  });

  it("renders the method steps", () => {
    render(<HybridDough />);
    const step = screen.queryAllByText(
      /Place the poolish in the fridge for between/i
    );
    expect(step).toBeDefined();
  });

  it("renders the alert with the correct description", () => {
    render(<HybridDough />);
    const alert = screen.queryAllByText(/to see if the dough is ready/i);
    expect(alert).toBeDefined();
  });
});
