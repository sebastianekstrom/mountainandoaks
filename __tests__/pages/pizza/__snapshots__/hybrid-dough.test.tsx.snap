// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HybridDough page renders the correct ingredients 1`] = `
<section
  class="mb-8 lg:mb-0"
  data-testid="INGREDIENT_LIST"
>
  <div>
     
  </div>
  <div
    class="mt-8"
  >
    <div
      class="mb-2"
    >
      <h3
        class="variable font-heading font-bold text-h3 leading-h3 text-regular "
      >
        Poolish
      </h3>
    </div>
    <div
      class="flex flex-col"
    >
      <ul>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Water
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            40
            g
          </p>
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Flour (Tipo 00)
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            40
            g
          </p>
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Honey
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            1
            g
          </p>
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Instant yeast
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            1
            g
          </p>
        </li>
      </ul>
      <div
        class="ml-auto"
      >
        <button
          class="mt-2 rounded-full text-regular bg-pill-background py-2 px-4"
          data-testid="BAKERS_PERCENTAGE"
          role="button"
          type="button"
        >
          <p
            class="variable font-sans text-caption leading-paragraph "
          >
            Show bakers %
          </p>
        </button>
      </div>
    </div>
  </div>
  <div
    class="mt-8"
  >
    <div
      class="mb-2"
    >
      <h3
        class="variable font-heading font-bold text-h3 leading-h3 text-regular "
      >
        Dough
      </h3>
    </div>
    <div
      class="flex flex-col"
    >
      <ul>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Manitoba flour
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            100
            g
          </p>
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Tipo 00 flour
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            30
            g
          </p>
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Water
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            65
            g
          </p>
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Sugar
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            13
            g
          </p>
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Olive oil
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            11
            g
          </p>
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Salt
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            5
            g
          </p>
        </li>
      </ul>
      <div
        class="ml-auto"
      >
        <button
          class="mt-2 rounded-full text-regular bg-pill-background py-2 px-4"
          data-testid="BAKERS_PERCENTAGE"
          role="button"
          type="button"
        >
          <p
            class="variable font-sans text-caption leading-paragraph "
          >
            Show bakers %
          </p>
        </button>
      </div>
    </div>
  </div>
</section>
`;
